Revamped Ignite Documentation (#1970)
infinitered/ignite: docs/Guide.md

In short -- if you use Ignite to start your next React Native project, you're using a battle-tested, familiar stack.

### Ignite CLI

In order to start a new Ignite project, you can use the CLI. No need to install it globally as it works great with `npx`:

```bash
npx ignite-cli new PizzaApp --expo
```

Once it's up and running, you can use the Ignite CLI to [generate](./Generators.md) components, screens, MST models, and more.

### Ignite Boilerplate

Your new Ignite project (whether you start with Expo or not) comes with a full stack of useful libraries, pre-set up for you so you can start coding.

- React Native
- React Navigation 6
- MobX-State-Tree [(Why not Redux?)](https://github.com/infinitered/ignite/blob/master/docs/MobX-State-Tree.md)
- MobX-React-Lite
- TypeScript
- AsyncStorage (integrated with MST for restoring state)
- apisauce (to talk to REST servers)
- Flipper-ready
- Reactotron-ready (and pre-integrated with MST)

> - Supports Expo (and Expo web) out of the box

Author: ```suggestion
- Supports Expo (and Expo web) out of the box
- About a dozen prebuilt [components](./Components.md) to build out your UI with
```

- Supports Expo (and Expo web) out of the box
- About a dozen prebuilt [components](./Components.md) to build out your UI with
- And more!

## Where do I start?

First, spin up the app and make sure you can see the welcome screen. If you have any issues, please report them.

Once it's running, you'll want to get familiarized with the following concepts:

### Navigation

We use React Navigation v6 in the current version of Ignite. You'll find any navigators in `./app/navigators`, with the `app-navigator.tsx` being the primary one.

There's also a `navigation-utilities.tsx` file which provides some utility functions we find useful in building apps, such as `getActiveRouteName`, `useBackButtonHandler` and `useNavigationPersistence`.

Learn more in our [Navigation](./Navigation.md) documentation.

### Components

Ignite comes with some prebuilt, flexible, and customizable components. Unlike most component libraries, it's not built to drop in out of the box, but rather with custom design in mind (you do have a nice, custom design, don't you?)

Ignite works fine with other component libraries, but the built-in component system works quite well for most apps.

> Check out the [Components][./components.md] documentation.

Author: ```suggestion
Check out the [Components](./components.md) documentation.
```

### Testing

Ignite is preconfigured to have [Detox](https://github.com/wix/Detox), Jest, and Storyshot tests.

You can learn more about why we chose these tests and how to use them in the [Testing](./Testing.md) docs.

### Styling

Ignite's approach to styling is, like many other things in Ignite, straightforward and blunt.

We don't use `StyleSheet.create()` as a general rule, as it doesn't provide any real benefits over bare objects.

> We instead use a strategy of constants, colocated with our components, prefixed with `$`, and typed with TypeScript:

Author: ```suggestion
We instead use a strategy of constants, colocated with our components, all upper-case, and typed with TypeScript:
```

```tsx
import { View, ViewStyle } from "react-native"
import { palette } from "../theme"

> const $container: ViewStyle = {

Author: ```suggestion
const CONTAINER: ViewStyle = {
```

  flex: 1,
  backgroundColor: palette.bgColor,
}

const MyComponent = () => {

>   return <View style={$container}>...</View>

Author: ```suggestion
  return <View style={CONTAINER}>...</View>
```
